app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,96 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
To show single post you need user permission ?

Reply:
Can we visit any post in Facebook without login?